server/templates/template_executor.go

```diff
@@ -63,12 +87,70 @@ func newJsTemplateExecutor(expression string, extraFunctions template.FuncMap) (
 			return nil, resError
 		}
 	}
-	//loads javascript into vm instance
-	function, err := LoadTemplateScript(script, extraFunctions)
+
+	jte := &asyncJsTemplateExecutor{make(chan events.Event), make(chan interface{}), script, nil}
+	go jte.start(extraFunctions)
+	_, err = jte.ProcessEvent(events.Event{})
+	if err != nil && strings.HasPrefix(err.Error(), jsLoadingErrorText) {
+		//we need to test that js function is properly loaded because that happens in asyncJsTemplateExecutor's goroutine
+		return nil, err
+	}
+	return jte, nil
+}
+
+
+func (jte *asyncJsTemplateExecutor) start(extraFunctions template.FuncMap) {
+	//loads javascript into new vm instance
+	function, err := LoadTemplateScript(jte.transformedExpression, extraFunctions)
 	if err != nil {
-		return nil, fmt.Errorf("js loading error: %v", err)
+		jte.loadingError =  fmt.Errorf("%s: %v\ntransformed function:\n%v\n",jsLoadingErrorText, err, jte.transformedExpression)
+	}
+	for {
```

Comment:
For such cases we have `safego` package that contains `safego.RunWithRestart()` func. Please take a look at the example https://github.com/jitsucom/jitsu/blob/master/server/adapters/http.go#L98

Reply:
done. but for entire start() function.